pkg/pillar/cmd/diag/diag.go

```diff
@@ -160,6 +162,26 @@ func Run(ps *pubsub.PubSub) {
 	log.Infof("Diag Get Device Serial %s, Soft Serial %s\n", zedcloudCtx.DevSerial,
 		zedcloudCtx.DevSoftSerial)
 
+	// Get UUID if available
+	for {
+		b, err := ioutil.ReadFile(types.UUIDFileName)
+		if err != nil {
+			log.Errorln("ReadFile", err, types.UUIDFileName)
+			break
+		}
```

Comment:
Would make more sense to Stat before reading; during onboarding the file does not exist.

If diag runs before client has onboarded (and created the uuid file), what will diag print?
Note that diag starts before client, and I don't see any code in diag to pick up a uuid file after it has started.
Thus diag will fail forever in the v2 onboarding scenario since it doesn't have the UUID in the URL? Or will get config URL work withoyt the UUID?

In the first case I think you need to have client publish the uuid as a persistent object in pubsub, and have diag subscribe to that.
In the second case you can just remove this code which gets the UUID.

Reply:
the cloud URL supports config with both using UUID and without UUID. this is also due to getUuid does a config, which does not have UUID at the time. so this is ok. Whenever the UUID is ready, it will use that.

Reply:
Let's create a story for the uuid being published by client (and used by diag etc). No need to do that in the next sprint but makes sense to track using a story.